src/main/java/pl/scoutCamp/api/controller/rest/RankingRestController.java

```diff
@@ -2,76 +2,43 @@
 
 import com.fasterxml.jackson.annotation.JsonView;
 import lombok.AllArgsConstructor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-import pl.scoutCamp.api.dto.TeamCategorizationSheetDTO;
-import pl.scoutCamp.api.dto.dtoList.TeamCategorizationSheetsDTO;
-import pl.scoutCamp.api.dto.mapper.TeamCategorizationSheetMapper;
-import pl.scoutCamp.business.TeamCategorizationSheetService;
-
-import java.util.List;
+import org.springframework.data.domain.Page;
+import org.springframework.web.bind.annotation.*;
+import pl.scoutCamp.api.dto.RankingDTO;
+import pl.scoutCamp.api.dto.dtoList.RankingsDTO;
+import pl.scoutCamp.api.dto.mapper.RankingMapper;
+import pl.scoutCamp.business.RankingService;
 
 @RestController
 @AllArgsConstructor
 @RequestMapping(RankingRestController.API_RANKING)
-
+// TODO: pagination and ranking in specific regiment
 public class RankingRestController {
 
     public static final String API_RANKING = "/ranking";
-    public static final String ALL_TEAMS = "/allTeams/{periodName}/sort_by={sortedBy}.{order}";
-    public static final String BY_REGIMENT = "/regiment={regimentId}/{periodName}/sort_by={sortedBy}.{order}";
+    public static final String ALL_TEAMS = "/allTeams/{period}";
 
-    private final TeamCategorizationSheetMapper teamCategorizationSheetMapper;
-    private final TeamCategorizationSheetService teamCategorizationSheetService;
+    private final RankingService rankingService;
+    private final RankingMapper rankingMapper;
 
     @GetMapping(value = ALL_TEAMS)
     @JsonView(JsonViews.RankingView.class)
-    public TeamCategorizationSheetsDTO findAllTeamsRanking
-            (
-                    @PathVariable String periodName,
-                    @PathVariable String sortedBy,
-                    @PathVariable String order
-            ) {
-        return getTeamCategorizationSheetsDTO(periodName, sortedBy, order);
-    }
-
-    @GetMapping(value = BY_REGIMENT)
-    @JsonView(JsonViews.RankingView.class)
-    public TeamCategorizationSheetsDTO findTeamsRankingByRegiment
-            (
-                    @PathVariable Integer regimentId,
-                    @PathVariable String periodName,
-                    @PathVariable String sortedBy,
-                    @PathVariable String order
-            ) {
-        return getTeamCategorizationSheetsDTO(regimentId, periodName, sortedBy, order);
+    public RankingsDTO findAllRankings (
+            @PathVariable String period,
+            @RequestParam(defaultValue = "points") String sort,
```

Comment:
What about organising these parameters in enums? To use specific values like SortType.desc instead of using strings directly? Is it proper Spring approach?